src/newgrf_gui.cpp

```diff
@@ -1314,10 +1320,29 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 		this->SetWidgetDisabledState(WID_NS_PRESET_SAVE, has_missing);
 	}
 
-	EventState OnKeyPress(WChar key, uint16 keycode) override
+	EventState OnKeyPressForActiveList(uint16 keycode)
 	{
-		if (!this->editable) return ES_NOT_HANDLED;
+		/* Get total actives and index of current selection. */
+		int active_pos = 0;
+		int active_count = 0;
+		for (GRFConfig *c = this->actives; c != nullptr; c = c->next, active_count++) {}
+		for (GRFConfig *c = this->actives; c != nullptr && c != this->active_sel; c = c->next, active_pos++) {}
 
+		if (this->vscroll2->UpdateListPositionOnKeyPress(active_pos, keycode) == ES_NOT_HANDLED) return ES_NOT_HANDLED;
+
+		active_pos = std::min(active_pos, active_count);
+
+		if (this->actives != nullptr) {
```

Comment:
The if-statement has no real effect. When `actives` is null, and you get here you have already clicked in the scrolled area so the active selection is already on 0. I would just remove the `if` and always run the code below.